custom_model_runner/datarobot_drum/resource/predict_mixin.py

```diff
@@ -47,7 +47,7 @@ def _validate_content_type_header(header):
         return ret_mimetype, ret_charset
 
     @staticmethod
-    def _fetch_data_from_request(file_key, logger=None):
+    def _fetch_data_from_request(file_key, logger=None, optional=False):
```

Comment:
I don't really like this construction with `optional` and  `elif len(request.data) and file_key == "X"; looks confusing.
Maybe add another method: `_fetch_additional_files_from_request`?